packages/plugin-kit/test/fs/writeCustomTypeModel.test.ts

```diff
@@ -0,0 +1,68 @@
+import { expect, it } from "vitest";
+import { createMockFactory } from "@prismicio/mock";
+import * as fs from "node:fs/promises";
+import * as path from "node:path";
+
+import { writeCustomTypeModel } from "../../src/fs";
+
+/**
+ * !!! DO NOT use this mock factory in tests !!!
+ *
+ * @remarks
+ * Its seed is not specific to be used outside the most general cases.
+ */
+const mock = createMockFactory({ seed: import.meta.url });
+const model = mock.model.customType();
+model.id = "foo_bar";
+
+it("saves a custom type's model", async (ctx) => {
+	await writeCustomTypeModel({
+		model,
+		helpers: ctx.pluginRunner.rawHelpers,
+	});
+
+	const contents = await fs.readFile(
+		path.join(ctx.project.root, "customtypes", model.id, "index.json"),
+		"utf8",
+	);
+
+	expect(JSON.parse(contents)).toStrictEqual(model);
+});
+
+it("formats contents if `format` is true", async (ctx) => {
+	await writeCustomTypeModel({
+		model,
+		format: true,
+		helpers: ctx.pluginRunner.rawHelpers,
+	});
+
+	const contents = await fs.readFile(
+		path.join(ctx.project.root, "customtypes", model.id, "index.json"),
+		"utf8",
+	);
+
+	expect(contents).not.toBe(JSON.stringify(model));
```

Comment:
I think this is always true since the model is still "lightly formatted" before being written: https://github.com/prismicio/slice-machine/blob/aa/plugin-kit-fs-helpers/packages/plugin-kit/src/fs/writeCustomTypeModel.ts#L29C1-L29C1, I guess we're looking at `expect(contents).not.toBe(JSON.stringify(model, null, 2);`

Might need to test with a dummy prettier config (e.g. tab instead of spaces)

Reply:
I'll finish this fix tomorrow!

Reply:
Fixed in https://github.com/prismicio/slice-machine/pull/1088/commits/d2e0f7ebb04e1692aeaf01787e03402fc4bbdaad. Thanks for the suggestion!